tree/criterion_interface.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef MACHINELEARNING_TREE_CRITERION_HPP
+#define MACHINELEARNING_TREE_CRITERION_HPP
+
+#include <math.h>
+
+namespace tree {
+
+/**
+ * @brief Base class that defines the many criterious of splitting.
+ *
+ * This interface provides some methods that are used to compute the impurity of a given Node object.
+ * Some impurity measures are Gini, Entrophy and Missclassification Error Rate for classification problens and MSE for regression
+ * problems.
+ */
+class CriterionInterface {
```

Comment:
It is pretty clear from the pure virtual methods that this is an `Interface` type class. You can safely drop the suffix.